oximeter/collector/src/lib.rs

```diff
@@ -79,12 +79,18 @@ pub struct DbConfig {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub address: Option<SocketAddr>,
 
-    /// Batch size of samples at which to insert
+    /// Batch size of samples at which to insert.
     pub batch_size: usize,
 
     /// Interval on which to insert data into the database, regardless of the number of collected
     /// samples. Value is in seconds.
     pub batch_interval: u64,
+
+    // TODO (https://github.com/oxidecomputer/omicron/issues/4148): This field
+    // should be removed if single node functionality is removed.
+    /// Whether ClickHouse is running as a replicated cluster or
+    /// single-node server.
```

Comment:
Oximeter has to initialize the database. It can't do that by asking whether it's talking to a replicated or single-node ClickHouse server, it has to _set up_ the server. Doesn't that mean we need some kind of information like this?

Reply:
Unless, based on your other comment, that `oximeter` isn't going to set up the databases in the long run. Is that the job of the `clickhouse-admin` server in the future? If so, then I agree with you :)

Reply:
Ah, I get it. I totally forgot about this dichotomy in the setup SQL. 

It could certainly be done via `clickhouse-admin`, but that was *not* my intent. I still think oximeter should likely run the sql to configure the database, but it should be told what to do by reconfigurator, especially when the old zone containing the single-node setup gets torn down and a replicated setup put in it's place. For now though, I guess that's all done via the oximeter config file. That negates a lot of my comments ;)

Reply:
Ok, that makes sense. One could imagine providing something like the same configuration that we give to `clickhouse-admin` to `oximeter` as well. For example, it already needs to register itself with Nexus. That request could provide information about which DB type to set up, for example, or `oximeter` could have an API through which Nexus / Reconfigurator provide that configuration.